src/code/z_collision_check.c

```diff
@@ -1,392 +1,3422 @@
+    return 1;
+}
 
-s32 func_8005B6EC(GlobalContext* globalCtx, Collider* collision, Actor* actor, ColliderBodyInfo* src) {
+// SetInit Collider (maskB = 0x10)
+s32 func_8005B6EC(GlobalContext* globalCtx, Collider* collision, Actor* actor, ColliderInit_set3* src) {
     collision->actor = actor;
     collision->unk_14 = src->unk_00;
-    collision->colliderFlags = src->colliderFlags;
-    collision->collideFlags = src->collideFlags;
+    collision->atFlags = src->atFlags;
+    collision->acFlags = src->acFlags;
     collision->maskA = src->maskA;
     collision->maskB = 0x10;
-    collision->type = src->maskB;
+    collision->shape = src->shape;
     return 1;
 }
 
-s32 func_8005B72C(GlobalContext* globalCtx, Collider* collision, Actor* actor, ColliderBodyInfo* src) {
+// SetInit Collider (maskB = src->maskB)
+s32 func_8005B72C(GlobalContext* globalCtx, Collider* collision, Actor* actor, ColliderInit* src) {
     collision->actor = actor;
     collision->unk_14 = src->unk_00;
-    collision->colliderFlags = src->colliderFlags;
-    collision->collideFlags = src->collideFlags;
+    collision->atFlags = src->atFlags;
+    collision->acFlags = src->acFlags;
     collision->maskA = src->maskA;
     collision->maskB = src->maskB;
-    collision->type = src->type;
+    collision->shape = src->shape;
     return 1;
 }
 
+// SetAT Collider base
 void func_8005B76C(GlobalContext* globalCtx, Collider* collision) {
     collision->at = NULL;
-    collision->colliderFlags &= ~0x6;
+    collision->atFlags &= ~0x6;
 }
 
+// SetAC Collider base
 void func_8005B784(GlobalContext* globalCtx, Collider* collision) {
     collision->ac = NULL;
-    collision->collideFlags &= ~0x82;
+    collision->acFlags &= ~0x82;
 }
 
+// SetOC Collider base
 void func_8005B79C(GlobalContext* globalCtx, Collider* collision) {
-    collision->ot = NULL;
+    collision->oc = NULL;
     collision->maskA &= ~0x2;
     collision->maskB &= ~0x1;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005B7C0.s")
+// Initialize ColliderTouch
+s32 func_8005B7C0(GlobalContext* globalCtx, ColliderTouch* touch) {
+    static ColliderTouch init = { 0, 0, 0 };
+    *touch = init;
+    return 1;
+}
 
+// Destruct ColliderTouch
 s32 func_8005B7E4(GlobalContext* globalCtx, ColliderTouch* touch) {
     return 1;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005B7F4.s")
+// SetInit ColliderTouch. src might actually be a different type
+s32 func_8005B7F4(GlobalContext* globalCtx, ColliderTouch* dest, ColliderTouch* src) {
+    dest->flags = src->flags;
+    dest->effect = src->effect;
+    dest->damage = src->damage;
+    return 1;
+}
 
 void func_8005B818(GlobalContext* globalCtx, ColliderBody* body) {
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005B824.s")
+// Initialize ColliderBump
+s32 func_8005B824(GlobalContext* globalCtx, ColliderBump* bump) {
+    static ColliderBump init = { (s32)0xFFCFFFFF, 0, 0, { 0, 0, 0 } };
```

Comment:
Is the (s32) cast necessary here? 
```suggestion
    static ColliderBump init = { 0xFFCFFFFF, 0, 0, { 0, 0, 0 } };
```